python/cudf_polars/cudf_polars/experimental/parallel.py

```diff
@@ -335,3 +337,26 @@ def _lower_ir_pwise(
 lower_ir_node.register(Cache, _lower_ir_pwise)
 lower_ir_node.register(HStack, _lower_ir_pwise)
 lower_ir_node.register(HConcat, _lower_ir_pwise)
+
+
+@lower_ir_node.register(Slice)
+def _(
+    ir: Slice, rec: LowerIRTransformer
+) -> tuple[IR, MutableMapping[IR, PartitionInfo]]:
+    if ir.offset == 0:
+        # Taking the first N rows.
+        # We don't know how large each partition is, so we reduce.
```

Comment:
Just to confirm my understanding of this code block: we're doing something like a `.slice(5)` to get the first 5 rows. We can't perform that on just the first partition, because it might not have 5 rows, so we need to repartition things.

Do we do the `.slice(5)` on each of the partitions, so that we aren't moving around full partitions? i.e. we know that the global `.slice(5)` can be written as something like `concat([x.slice(5) for x in partitions]).slice(5)`.